uc.c

```diff
@@ -2394,6 +2419,31 @@ gint cmp_vaddr(gconstpointer a, gconstpointer b, gpointer user_data)
     return va < vb ? -1 : 1;
 }
 
+UNICORN_EXPORT
+uc_err uc_snapshot(struct uc_struct *uc)
```

Comment:
If I understand correctly, the reason why you still need an extra API though `uc_context_*` already include the functionality is that you want to save memory but not CPU context, right? If so, I suppose we use CTL to specify what we need to save, e.g. a new CTL named like `UC_CTL_CONTEXT_CONTENT` and the possible values are like `UC_CTL_CONTEXT_CPU | UC_CTL_CONTEXT_MEMORY` etc. `snapshot` looks almost identical to `context` and it's not good to expose a separate API here AFAIK.

Reply:
Sounds good. I'll look at this a bit more and probably remove the `uc_snapshot` api.